polygeist/lib/Dialect/Polygeist/Utils/TransformUtils.cpp

```diff
@@ -242,11 +255,10 @@ void FunctionKernelInfo::populateGPUKernelInfo(FunctionOpInterface func) {
   funcKernelInfosMap[func] = {};
 
   Operation *op = func;
-  if (auto gpuFunc = dyn_cast<gpu::GPUFuncOp>(op))
-    if (gpuFunc.isKernel()) {
-      funcKernelInfosMap[func].push_back({gpuFunc, 0});
-      return;
-    }
+  if (auto gpuFunc = dyn_cast<gpu::GPUFuncOp>(op); gpuFunc.isKernel()) {
```

Comment:
I think with this change, `gpuFunc.isKernel` is executed no matter `op` is `GPUFuncOp` or not, so it may segfault?